pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -369,7 +352,10 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 	}
 	availableCurrent -= toDel
 
-	switch drift := reserveSize - int(clp.Spec.Size); {
+	// drift will indicate how many clusters we need to add or delete to get back to steady state
+	// of the pool's Size. This needs to take into account the clusters we're creating to satisfy
+	// the immediate demand of pending claims.
+	switch drift := len(cds.Unassigned(true)) - int(clp.Spec.Size) - len(claims.Unassigned()); {
```

Comment:
Question about original implementation - drift is calculated considering broken clusters, but we delete them before entering case of drift==0 - shouldn't drift be recalculated then? Maybe we need to add some clusters?